radix-engine-interface/src/blueprints/resource/bucket.rs

```diff
@@ -8,206 +8,56 @@ use crate::*;
-pub struct BucketPutInvocation {
-    pub receiver: BucketId,
+pub struct BucketPutInput {
     pub bucket: Bucket,
 }
 
-impl Clone for BucketPutInvocation {
+impl Clone for BucketPutInput {
     fn clone(&self) -> Self {
         Self {
-            receiver: self.receiver,
             bucket: Bucket(self.bucket.0),
         }
     }
 }
 
-impl Invocation for BucketPutInvocation {
-    type Output = ();
-
-    fn fn_identifier(&self) -> FnIdentifier {
-        FnIdentifier::Native(NativeFn::Bucket(BucketFn::Put))
-    }
-}
-
-impl SerializableInvocation for BucketPutInvocation {
-    type ScryptoOutput = ();
-
-    fn native_fn() -> NativeFn {
-        NativeFn::Bucket(BucketFn::Put)
-    }
-}
-
-impl Into<CallTableInvocation> for BucketPutInvocation {
-    fn into(self) -> CallTableInvocation {
-        NativeInvocation::Bucket(BucketInvocation::Put(self)).into()
-    }
-}
+pub const BUCKET_TAKE_NON_FUNGIBLES_IDENT: &str = "Bucket_take_non_fungibles";
 
 #[derive(Debug, Clone, Eq, PartialEq, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
-pub struct BucketTakeNonFungiblesInvocation {
-    pub receiver: BucketId,
+pub struct BucketTakeNonFungiblesInput {
     pub ids: BTreeSet<NonFungibleLocalId>,
 }
 
-impl Invocation for BucketTakeNonFungiblesInvocation {
-    type Output = Bucket;
-
-    fn fn_identifier(&self) -> FnIdentifier {
-        FnIdentifier::Native(NativeFn::Bucket(BucketFn::TakeNonFungibles))
-    }
-}
-
-impl SerializableInvocation for BucketTakeNonFungiblesInvocation {
-    type ScryptoOutput = Bucket;
-
-    fn native_fn() -> NativeFn {
-        NativeFn::Bucket(BucketFn::TakeNonFungibles)
-    }
-}
-
-impl Into<CallTableInvocation> for BucketTakeNonFungiblesInvocation {
-    fn into(self) -> CallTableInvocation {
-        NativeInvocation::Bucket(BucketInvocation::TakeNonFungibles(self)).into()
-    }
-}
+pub const BUCKET_GET_NON_FUNGIBLE_LOCAL_IDS_IDENT: &str = "Bucket_get_non_fungible_local_ids";
 
 #[derive(Debug, Clone, Eq, PartialEq, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
-pub struct BucketGetNonFungibleLocalIdsInvocation {
-    pub receiver: BucketId,
-}
+pub struct BucketGetNonFungibleLocalIdsInput {}
 
-impl Invocation for BucketGetNonFungibleLocalIdsInvocation {
-    type Output = BTreeSet<NonFungibleLocalId>;
-
-    fn fn_identifier(&self) -> FnIdentifier {
-        FnIdentifier::Native(NativeFn::Bucket(BucketFn::GetNonFungibleLocalIds))
-    }
-}
-
-impl SerializableInvocation for BucketGetNonFungibleLocalIdsInvocation {
-    type ScryptoOutput = BTreeSet<NonFungibleLocalId>;
-
-    fn native_fn() -> NativeFn {
-        NativeFn::Bucket(BucketFn::GetNonFungibleLocalIds)
-    }
-}
-
-impl Into<CallTableInvocation> for BucketGetNonFungibleLocalIdsInvocation {
-    fn into(self) -> CallTableInvocation {
-        NativeInvocation::Bucket(BucketInvocation::GetNonFungibleLocalIds(self)).into()
-    }
-}
+pub const BUCKET_GET_AMOUNT_IDENT: &str = "Bucket_get_amount";
```

Comment:
Wait, why do we start adding `Bucket_` prefix here? Grouping exports of different blueprints within the same package?

I feel like we might want consistent `export_name` for all our native blueprints, or split it into `blueprint_ident + fn_ident`.

Reply:
Yup agreed.

Reply:
Can we push this to a later PR as would like to put this together into a large PR.